### PR TITLE
Limited search endpoint.

### PR DESCRIPTION
**Pull request by dianakhuang**
Add a stripped down endpoint to be consumed by Prospectus.

https://openedx.atlassian.net/browse/WEBSITE-64

### CODE REVIEW

File: course_discovery/apps/api/v1/urls.py [hunk 1]
```diff
@@ -45,6 +45,7 @@
 router.register(r'pathways', PathwayViewSet, basename='pathway')
 router.register(r'programs', ProgramViewSet, basename='program')
 router.register(r'program_types', ProgramTypeViewSet, basename='program_type')
+router.register(r'search/all/limited', search_views.LimitedAggregateSearchView, basename='search-all-limited')
```
Comment (rgraber) [inline course_discovery/apps/api/v1/urls.py]: "all" and "limited" seem like a contradiction in terms, i'd vote for just search/limited

File: course_discovery/apps/api/v1/tests/test_views/test_search.py [hunk 2]
```diff
@@ -364,6 +364,79 @@ def test_results_include_aggregation_key(self):
...
+
+        if query:
+            qs = urllib.parse.urlencode(query)
+
+        url = '{path}?{qs}'.format(path=self.path, qs=qs)
+        return self.client.get(url)
+
+    def process_response(self, response):
+        response = self.get_response(response).json()
+        print(response)
```
Comment (rgraber) [inline course_discovery/apps/api/v1/tests/test_views/test_search.py]: extraneous print

File: course_discovery/apps/api/v1/tests/test_views/test_search.py [hunk 3]
```diff
@@ -364,6 +364,79 @@ def test_results_include_aggregation_key(self):
...
+        hidden_run = CourseRunFactory(course__partner=self.partner, hidden=True)
+
+        assert CourseRun.objects.get(hidden=True) == hidden_run
+
+        response = self.get_response()
+        data = response.json()
+        assert data['objects']['results'] == [self.serialize_course_run_search(visible_run)]
+
+    def test_results_include_aggregation_key(self):
+        """ Verify the search results only include the aggregation_key for each document. """
```
Comment (rgraber) [inline course_discovery/apps/api/v1/tests/test_views/test_search.py:409]: i don't totally understand what this is testing. only get the aggregation key as opposed to what?

File: course_discovery/apps/api/v1/tests/test_views/test_search.py [hunk 4]
```diff
@@ -364,6 +364,79 @@ def test_results_include_aggregation_key(self):
         assert expected == actual
 
 
+class LimitedAggregateSearchViewSetTests(
```
Comment (rgraber) [inline course_discovery/apps/api/v1/tests/test_views/test_search.py:367]: can we test that we don't get any School/Org objects?

File: course_discovery/apps/api/v1/tests/test_views/test_search.py [hunk 5]
```diff
@@ -364,6 +364,79 @@ def test_results_include_aggregation_key(self):
...
+
+    # pylint: disable=no-member
+    def serialize_course_run_search(self, run):
+        return super().serialize_course_run_search(run, serializers.LimitedAggregateSearchModelSerializer)
+
+    # pylint: disable=no-member
+    def serialize_program_search(self, program):
+        return super().serialize_program_search(program, serializers.LimitedAggregateSearchModelSerializer)
+
+    def get_response(self, query=None):
```
Comment (rgraber) [inline course_discovery/apps/api/v1/tests/test_views/test_search.py]: query param is unused in tests

File: course_discovery/apps/api/v1/tests/test_views/test_search.py [hunk 6]
```diff
@@ -364,6 +364,79 @@ def test_results_include_aggregation_key(self):
...
+        qs = ''
+
+        if query:
+            qs = urllib.parse.urlencode(query)
+
+        url = '{path}?{qs}'.format(path=self.path, qs=qs)
+        return self.client.get(url)
+
+    def process_response(self, response):
```
Comment (rgraber) [inline course_discovery/apps/api/v1/tests/test_views/test_search.py]: unused method

**Review by rgraber — COMMENTED** [review]
A few small questions but the approach looks good. I don't know haystack too well, I'm assuming this will only search the Course/Run/Programs indices and not search everything and then just filter them out before display?

File: course_discovery/apps/api/v1/tests/test_views/test_search.py (hunk 4 above)
Comment (dianakhuang) [reply course_discovery/apps/api/v1/tests/test_views/test_search.py:367]: It doesn't seem likely given that we're not actually indexing any of them.

**Review by rgraber — APPROVED** [review]
LGTM!

**Pull request merged**